[mlir] make mlir related stuff build on macos

freedomtan opened the pull request:
`bazel build --config opt tensorflow/compiler/mlir/...` doesn't work on macOS.
1. `exp10()` is not a standard C function, macos has `__exp10()`
2. include `"mlir/StandardOps/Ops.h"` earlier to avoid `TRUE` and`FALSE` being defined in `<mach/boolean.h>`

freedomtan committed:
make mlir related stuff build on macos
1. exp10() is not a standard C function, macos has __exp10()
2. include "mlir/StandardOps/Ops.h" earlier to avoid TRUE and
   FALSE being defined in <mach/boolean.h>

jpienaar commented on tensorflow/compiler/mlir/lite/ir/tfl_traits.h:
```diff
@@ -22,6 +22,11 @@ limitations under the License.
 #include "mlir/Support/LLVM.h"  // TF:local_config_mlir
 #include "tensorflow/compiler/mlir/lite/utils/quantization_utils.h"
 
+// exp10() is not a standard function
```
Could you update the call site to use pow instead?

freedomtan replied:
sure.

jpienaar commented:
One suggestion, else it looks good, thanks

freedomtan committed:
replace exp10(X) with portable standard pow(10.0, X)

freedomtan commented:
rebased to resolve conflicts and replaced `exp10(X)` with `pow(10.0, X)`

jpienaar reviewed (dismissed):
Looks good, I've also updated TRUE & FALSE to avoid needing to make the non-pow changes.

freedomtan commented:
Thanks for your work and review. BTW, I cannot build related stuff after 22385cc65e2, either reverting it or more modifications needed.

jpienaar commented:
Could you rebase? We've changed function to be value typed. If that doesn't work, I'll apply your change to head and commit it.

freedomtan commented:
Yes, 604988b works. The only change needed is replacing `exp10()`. I'll update with additional `"mlir/StandardOps/Ops.h"`removed.

freedomtan committed:
revert "mlir/StandardOps/Ops.h" chagnes

remove additional "mlir/StandardOps/Ops.h" chagnes since there is
no boolean problem anymore
